### PR TITLE
Docs auth retries

### PR DESCRIPTION
**Pull request by Nadine2016**
# Description

Please include a high level summary of the changes.

This bug fixes ... \ This new feature can be used to ...

_Fill out any of the following sections that are relevant and remove the others_
## API changes
- Added x field to y resource
- ...

## Code changes
- Fix error in `Foo()` function
- Add `Bar()` function
- ...

## CI changes
- Adjusted schedule for x job
- ...

## Docs changes
- Added guide about feature x to public docs
- Updated README to account for y behavior
- ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

See slack conversation [here](https://solo-io-corp.slack.com/archives/some/post)

## Interesting decisions
 
We chose to do things this way because ...

## Testing steps

I manually verified behavior by ...

## Notes for reviewers

Be sure to verify intended behavior by ...

Please proofread comments on ...

This is a complex PR and may require a huddle to discuss ...

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/security/auth/extauth/passthrough_auth/grpc/_index.md [hunk 1]
```diff
@@ -210,6 +210,40 @@ curl -H "Host: foo" -H "authorization: authorize me" $(glooctl proxy url)/posts/
...
+      grpc:
+        # Address of the grpc auth server to query
+        address: example-grpc-auth-service.default.svc.cluster.local:9001
+        # Set a connection timeout to external service, default is 5 seconds
+        connectionTimeout: 3s
+        retryPolicy: 
+          numRetries: 10
+          retryBackOff:
+            baseInterval: 1s
+            maxInterval: 2ms
```
Comment (davidjumani) [inline docs/content/guides/security/auth/extauth/passthrough_auth/grpc/_index.md:240]:
```suggestion
            maxInterval: 2s
```

File: docs/content/guides/security/auth/extauth/passthrough_auth/grpc/_index.md [hunk 2]
```diff
@@ -210,6 +210,40 @@ curl -H "Host: foo" -H "authorization: authorize me" $(glooctl proxy url)/posts/
...
+
+
+* **Passthrough service becomes unavailable after initial connection**: In this scenario, the Gloo ExtAuth server successfully established an initial connection to the passthrough service. However later on, the passthrough service becomes unavailable. You can add a retry policy to your AuthConfig to retry the connection to the passthrough service if the service becomes unavailable. In the following AuthConfig, the Gloo ExtAuth server is configured to retry the connection to the passthrough service 10 times. To not overload the passthrough service, an optional exponential backoff strategy is defined. The backoff strategy configures the ExtAuth server to start retries after 1 second (`baseInterval`). Retries are then executed exponentially, such as after 2 seconds, 4 seconds, 8 seconds, etc. Note that retries are only executed up to the defined `global.extensions.extAuth.requestTimeout` which defaults to 10 seconds. The `maxInterval` configures a maximum delay of 2 seconds between retries.
```
Comment (davidjumani) [inline docs/content/guides/security/auth/extauth/passthrough_auth/grpc/_index.md:218]:
```suggestion
* **Passthrough service becomes unavailable after initial connection**: In this scenario, the Gloo ExtAuth server successfully established an initial connection to the passthrough service. However later on, the passthrough service becomes unavailable. You can add a retry policy to your AuthConfig to retry the connection to the passthrough service if the service becomes unavailable. In the following AuthConfig, the Gloo ExtAuth server is configured to retry the connection to the passthrough service 10 times. To not overload the passthrough service, an optional exponential backoff strategy is defined. The backoff strategy configures the ExtAuth server to start retries after 1 second (`baseInterval`). Retries are then executed exponentially, such as after 2 seconds, 4 seconds, 8 seconds, etc up to the `maxInterval` that defaults to 10 times the `baseInterval`. In this example, the `maxInterval` configures a maximum delay of 2 seconds between retries. Note that the global settings `global.extensions.extAuth.requestTimeout` must be greater than the `retryPolicy.numRetries` * `retryPolicy.retryBackOff.baseInterval` to ensure that the failed gRPC call has sufficient time to retry.
```

**Review by davidjumani — CHANGES_REQUESTED** [review]
Changes regarding the use of `global.extensions.extAuth.requestTimeout`

**Review by davidjumani — APPROVED** [review]

**Review by artberger — APPROVED** [review]

**Pull request merged**